sql: Add information_schema.session_variables table

mneverov opened the pull request:
sql: Add information_schema.session_variables table

The `session_variables` table exposes the session variables.
Related to: #8675

Release note (sql change): Added session_variables table to the information_schema.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/57837)
<!-- Reviewable:end -->

mneverov reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/57837)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @jordanlewis, @nvanbenschoten, @otan, and @rafiss)

*[a discussion](https://reviewable.io/reviews/cockroachdb/cockroach/57837#-MOIF4jN3beOi5GcYJpJ:-MOIF4jO9AS7l-7zX8pJ:b-nkv3l2) (no related file):*
To the reviewer: it appears CRDB does already contain this information in [crdb_internal.session_variables](https://github.com/cockroachdb/cockroach/blob/2b897f3007af0b0de0783535a4e62788d2ab6a62/pkg/sql/crdb_internal.go#L1184).
This commit is just a copy-paste from the crdb_internal.session_variables without `hidden` column.
Please let me know if hidden variables should be filtered out: allow_prepare_as_opt_plan, save_tables_prefix, session_authorization, ssl_renegotiation_limit

---


<!-- Sent from Reviewable.io -->

mneverov committed:
sql: Add information_schema.session_variables table

The `session_variables` table exposes the session variables.
Related to: #8675

Release note (sql change): Added session_variables table to the information_schema.

rafiss reviewed (approved):
thanks for your contribution!

bors r+
